pkg/controller/pxc/controller.go

```diff
@@ -1084,7 +1084,12 @@ func (r *ReconcilePerconaXtraDBCluster) createOrUpdate(obj runtime.Object) error
 	objAnnotations["percona.com/last-config-hash"] = hash
 	objectMeta.SetAnnotations(objAnnotations)
 
-	oldObject := obj.DeepCopyObject()
+	val := reflect.ValueOf(obj)
```

Comment:
But deepcopy functions are creating new objects
for example
 https://github.com/percona/percona-xtradb-cluster-operator/blob/main/pkg/apis/pxc/v1/zz_generated.deepcopy.go#L670
 
 ```
 z := new(PerconaXtraDBClusterSpec)
y := z.DeepCopy()
fmt.Println(z == y)
fmt.Println(&z)
fmt.Println(&y)
-------------------------------
false
0xc0000b0cb8
0xc0000b0cc0
```

Reply:
Yes, you right. Someway I thought it's because of pointers.
But in any case with `DeepEqual` this function does not work